### PR TITLE
Handle ChannelFuture null scenario in NetworkClient.closeAsync()

### PR DESCRIPTION
**Pull request by smithakoduri**

### CODE REVIEW

File: riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java [hunk 1]
```diff
@@ -114,18 +114,22 @@ public void closeAsync() {
...
+                            }
                         }
-                    }
-                });
+                    });
+                    return;
+                }
             }
         }
+        shutdown();
```
Comment (ymatsuda) [inline riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java:132]: `shutdown()` shouldn't be called when `isValid()` is false.

Comment (smithakoduri) [reply riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java]: fixed.

File: riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java [hunk 2]
```diff
@@ -124,6 +123,13 @@ public void closeAsync() {
                         }
                     }
                 });
+                return;
+            }
+        }
+        synchronized (state) {
+            if (isValid()) {
```
Comment (ymatsuda) [inline riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java:130]: Why not putting this in `synchronized (this) { ... }`?

Comment (smithakoduri) [reply riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java:111]: Fixed. Based on the discussion, modified the code in NetworkClient to use synchronized (this) and remove synchronized (state).

**Review by ymatsuda — APPROVED** [review]

**Pull request merged**